File: src/csharp/Intel/Iced/Intel/CodeReader.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if DECODER
namespace Iced.Intel {
	/// <summary>
	/// Reads instruction bytes
	/// </summary>
	public abstract class CodeReader {
		/// <summary>
		/// Reads the next byte or returns less than 0 if there are no more bytes
		/// </summary>
		/// <returns></returns>
		public abstract int ReadByte();
	}
}
#endif

File: src/java/iced-x86/src/main/java/com/github/icedland/iced/x86/dec/CodeReader.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.dec;

/**
 * Reads instruction bytes
 */
public abstract class CodeReader {
	/**
	 * Reads the next byte ({@code 0x00-0xFF}) or returns less than 0 if there are no more bytes
	 */
	public abstract int readByte();
}
